packages/scene-composer/src/components/panels/scene-settings/OverlayPanelVisibilityToggle.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useContext, useEffect } from 'react';
+import { useIntl } from 'react-intl';
+
+import { useStore, useViewOptionState } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { IComponentSettingsMap, IOverlaySettings, KnownComponentType, KnownSceneProperty } from '../../../interfaces';
+import { Component } from '../../../models/SceneModels';
+import { componentSettingsSelector } from '../../../utils/componentSettingsUtils';
+
+import { ComponentVisibilityToggle } from './ComponentVisibilityToggle';
+
+export const OverlayPanelVisibilityToggle: React.FC = () => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const { formatMessage } = useIntl();
+  const overlayPanelVisible =
+    useViewOptionState(sceneComposerId).componentVisibilities[Component.DataOverlaySubType.OverlayPanel];
+  const toggleComponentVisibility = useViewOptionState(sceneComposerId).toggleComponentVisibility;
+
+  const documentSettings: IOverlaySettings = useStore(sceneComposerId)(
+    (state) => componentSettingsSelector(state, KnownComponentType.DataOverlay) as IOverlaySettings,
+  );
+
+  const setSceneProperty = useStore(sceneComposerId)((state) => state.setSceneProperty);
+  const getSceneProperty = useStore(sceneComposerId)((state) => state.getSceneProperty);
+  const isViewing = useStore(sceneComposerId)((state) => state.isViewing());
+
+  // When the document settings is changed from other places (e.g. undo/redo),
+  // sync the view options visibility value with document settings.
+  useEffect(() => {
+    if (!isViewing && !!overlayPanelVisible !== !!documentSettings.overlayPanelVisible) {
```

Comment:
Too many checks in one condition. May be you can simply this logic.

Reply:
There's only two checks here. What are you referencing for this being too many?

Reply:
1. !isViewing
2. !!overlayPanelVisible
3. !!documentSettings.overlayPanelVisible
4. !!overlayPanelVisible !== !!documentSettings.overlayPanelVisible
5. !isViewing && !!overlayPanelVisible !== !!documentSettings.overlayPanelVisible

Reply:
2 & 3 are just to convert nullable booleans to boolean type because for this code, `undefined` and `false` are the same. so there are just 2 conditions here

Reply:
I am not talking about conditions but talking about converting these values or checking the state of the each values part of these two conditions. It would be good if the logics are easy to read.